### PR TITLE
adding resource bundling for acme and reloader

### PR DESCRIPTION
**Pull request by mubariskhan96**
This is a draft PR 

I need to test it and do the thing that I have done for two operators, for several other operators. 

Comments welcome

### CODE REVIEW

**Comment by larsks** [comment]
This looks like you're heading in the right direction! You'll want to test these changes to make sure they produce the same configuration as the original source.  Unfortunately, we're storing some encrypted data in this repository which is normally processed with the [ksops](https://github.com/viaduct-ai/kustomize-sops)  plugin, which means if you try to run `kustomize build` in an overlay directory, you'll see an error like:

> Error: accumulating resources: accumulation err='accumulating resources from '../common': '/home/lars/projects/operate-first/apps/cluster-scope/overlays/prod/moc/common' must resolve to a file': recursed accumulation of path '/home/lars/projects/operate-first/apps/cluster-scope/overlays/prod/moc/common': loading generator plugins: external plugins disabled; unable to load external plugin 'ksops'

To deal with this error, we need to do three things:

1. Tell `kustomize` to enable plugins with the `--enable-alpha-plugins` flag.
2. Create a "fake" version of the ksops plugin so that you can test things out without requiring a decryption key.
3. Tell `kustomize` where to find our fake plugin

To create the fake plugin:

1. At the top level of the repository, create the directory `plugins/viaduct.ai/v1/ksops`.
2. Symlink `plugins/viaduct.ai/v1/ksops` to `/bin/true`:

    ```
    ln -s /bin/true plugins/viaduct.ai/v1/ksops/ksops
    ```

To tell `kustomize` where to find this plugin, we'll need to set the
`KUSTOMIZE_PLUGIN_HOME` environment variable to `$PWD/plugins`.

That whole process should look like:

```
$ cd apps
$ mkdir -p plugins/viaduct.ai/v1/ksops
$ ln -s /bin/true plugins/viaduct.ai/v1/ksops/ksops
$ export KUSTOMIZE_PLUGIN_HOME=$PWD/plugins
$ cd cluster-scope/overlays/prod/moc/smaug
$ git checkout master
$ kustomize build --enable-alpha-plugins > out-master.yaml
$ git checkout resourceBundling
$ kustomize build --enable-alpha-plugins > out-bundling.yaml
$ diff -u out-master.yaml out-bundling.yaml
```

You should not see any output from the `diff` command. If you do, make
sure your branch is up-to-date with respect to the `master` branch.

Lastly, please don't just copy and paste any of these commands: if you
don't understand what something does, please ask!

**Comment by mubariskhan96** [comment]
Hi @larsks thanks for the step by step commands. I'm working through the commands, but seems like there's some issue with the plugin. 
When I run the `kustomize build` command, I get the following error. 

`Error: accumulating resources: accumulation err='accumulating resources from '../common': '/home/mubariskhan/Desktop/redhat/apps/cluster-scope/overlays/prod/moc/common' must resolve to a file': recursed accumulation of path '/home/mubariskhan/Desktop/redhat/apps/cluster-scope/overlays/prod/moc/common': loading generator plugins: unable to find plugin root - tried: ('<no value>'; homed in $KUSTOMIZE_PLUGIN_HOME), ('<no value>'; homed in $XDG_CONFIG_HOME), ('/root/.config/kustomize/plugin'; homed in default value of $XDG_CONFIG_HOME), ('/root/kustomize/plugin'; homed in home directory)
`
I'm trying to see what might have gone wrong, and upon checking, I do have the `plugins/viaduct.ai/v1/ksops` directory created . Also, it looks like the symlink is also set ( I ran the command to create the symlink again and it said, it already exists). 

Also, from [this](https://wdicc.com/kustomize-plugins/) article, it seems like, kustomize is not able to find the plugin. So, probably, `export KUSTOMIZE_PLUGIN_HOME=$PWD/plugins` this command is either not working correctly, or I'm not doing something right there. But, I'm not sure what I'm missing. 

Can you please elaborate on `export KUSTOMIZE_PLUGIN_HOME=$PWD/plugins` this command? 
I suspect my kustomize is not able to get this right. 

Thanks!

**Comment by larsks** [comment]
@mubariskhan96 let's take a look at this after the group meeting this afternoon.

**Comment by mubariskhan96** [comment]
@larsks Great news! 

`diff -u out-master.yaml out-bundling.yaml`

Returned nothing, so that means what I've got so far is correct. So, I think this PR can be merged. I will create a separate PR for the other operators. @HumairAK @4n4nd

File: cluster-scope/bundles/b4mad-racing/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+resources:
+  - ../../base/core/namespaces/b4mad-racing
```
Comment (larsks) [inline cluster-scope/bundles/b4mad-racing/kustomization.yaml:5]: We shouldn't create single-resource bundles. This should just be included directly from the `kustomization.yaml`.

**Review by larsks — CHANGES_REQUESTED** [review]

File: cluster-scope/bundles/b4mad-racing/kustomization.yaml (hunk 1 above)
Comment (mubariskhan96) [reply cluster-scope/bundles/b4mad-racing/kustomization.yaml]: oops! Thanks for the feedback, I have updated the PR

**Review by HumairAK — COMMENTED** [review]
The `fybrik` stuff also looks like it could be grouped, skimming through -- these seem like they ought to be grouped together: 

```
- ../../../../base/apiextensions.k8s.io/customresourcedefinitions/assets.katalog.fybrik.io
- ../../../../base/apiextensions.k8s.io/customresourcedefinitions/batchtransfers.motion.fybrik.io
- ../../../../base/apiextensions.k8s.io/customresourcedefinitions/blueprints.app.fybrik.io
- ../../../../base/apiextensions.k8s.io/customresourcedefinitions/extensions.dashboard.tekton.dev
- ../../../../base/apiextensions.k8s.io/customresourcedefinitions/fybrikapplications.app.fybrik.io
- ../../../../base/apiextensions.k8s.io/customresourcedefinitions/fybrikmodules.app.fybrik.io
- ../../../../base/apiextensions.k8s.io/customresourcedefinitions/fybrikstorageaccounts.app.fybrik.io
- ../../../../base/apiextensions.k8s.io/customresourcedefinitions/plotters.app.fybrik.io
- ../../../../base/apiextensions.k8s.io/customresourcedefinitions/prowjobs.prow.k8s.io
- ../../../../base/apiextensions.k8s.io/customresourcedefinitions/reports.batch.curator.openshift.io
- ../../../../base/apiextensions.k8s.io/customresourcedefinitions/streamtransfers.motion.fybrik.io
- ../../../../base/core/namespaces/fybrik-applications
- ../../../../base/core/namespaces/fybrik-blueprints
- ../../../../base/core/namespaces/fybrik-system
- ../../../../base/rbac.authorization.k8s.io/clusterrolebindings/fybrik-admin-rb
- ../../../../base/rbac.authorization.k8s.io/clusterroles/fybrik-admin-cr
```

Maybe we can get input from @eletonia here. For context @eletonia we're working towards bundling manifests in this large kustomization. Manifests from the `base` folder that ought to be deployed together should be categorized as a "bundle", so that we can track them more easily, and the main `kustomization` is not as bloated. Can you confirm that the manifests above go together? Feel free to propose a better grouping.

File: cluster-scope/overlays/prod/moc/smaug/kustomization.yaml [hunk 2]
```diff
@@ -27,7 +33,6 @@ resources:
 - ../../../../base/config.openshift.io/oauths/cluster
 - ../../../../base/core/configmaps/cluster-monitoring-config
 - ../../../../base/core/configmaps/user-workload-monitoring-config
-- ../../../../base/core/namespaces/acme-operator
```
Comment (HumairAK) [inline cluster-scope/overlays/prod/moc/smaug/kustomization.yaml:30]: Have you also checked the other kustomizations? `smaug` is not the only cluster that uses these manifests from the `base` directory. 

`acme-operator` for instance is also used by the osc cluster, so it's included in the osc overlay [here](https://github.com/operate-first/apps/blob/master/cluster-scope/overlays/prod/osc/osc-cl1/kustomization.yaml).

Comment (HumairAK) [reply cluster-scope/overlays/prod/moc/smaug/kustomization.yaml:30]: You should cross check `diffs` from  buildling all the kustomizations that occur in `apps/cluster-scope/overlays/prod/$ENV/$CLUSTER/`

**Comment by eletonia** [comment]
> The `fybrik` stuff also looks like it could be grouped, skimming through -- these seem like they ought to be grouped together:
> 
> ```
> - ../../../../base/apiextensions.k8s.io/customresourcedefinitions/assets.katalog.fybrik.io
> - ../../../../base/apiextensions.k8s.io/customresourcedefinitions/batchtransfers.motion.fybrik.io
> - ../../../../base/apiextensions.k8s.io/customresourcedefinitions/blueprints.app.fybrik.io
> - ../../../../base/apiextensions.k8s.io/customresourcedefinitions/fybrikapplications.app.fybrik.io
> - ../../../../base/apiextensions.k8s.io/customresourcedefinitions/fybrikmodules.app.fybrik.io
> - ../../../../base/apiextensions.k8s.io/customresourcedefinitions/fybrikstorageaccounts.app.fybrik.io
> - ../../../../base/apiextensions.k8s.io/customresourcedefinitions/plotters.app.fybrik.io
> - ../../../../base/apiextensions.k8s.io/customresourcedefinitions/streamtransfers.motion.fybrik.io
> - ../../../../base/core/namespaces/fybrik-applications
> - ../../../../base/core/namespaces/fybrik-blueprints
> - ../../../../base/core/namespaces/fybrik-system
> - ../../../../base/rbac.authorization.k8s.io/clusterrolebindings/fybrik-admin-rb
> - ../../../../base/rbac.authorization.k8s.io/clusterroles/fybrik-admin-cr
> ```
> 
> Maybe we can get input from @eletonia here. For context @eletonia we're working towards bundling manifests in this large kustomization. Manifests from the `base` folder that ought to be deployed together should be categorized as a "bundle", so that we can track them more easily, and the main `kustomization` is not as bloated. Can you confirm that the manifests above should be bundled together? Feel free to propose a better grouping.

@HumairAK Yes all of these manifests can be bundled - all of these Fybrik resources should be deployed together @roytman

**Comment by HumairAK** [comment]
Can we also update the docs here along with the PR on what the bundles mean and incorporate it as part of the structure? 
https://github.com/operate-first/apps/tree/master/cluster-scope

File: cluster-scope/overlays/prod/moc/smaug/kustomization.yaml (hunk 2 above)
Comment (mubariskhan96) [reply cluster-scope/overlays/prod/moc/smaug/kustomization.yaml:28]: Thanks for taking a look! In this PR, I'm only doing bundling for smaug cluster. I will be creating more PRs to send in the changes for other clusters.

**Comment by mubariskhan96** [comment]
> Can we also update the docs here along with the PR on what the bundles mean and incorporate it as part of the structure? https://github.com/operate-first/apps/tree/master/cluster-scope

okay, I was trying to make a relatively small PR. Would it be a good idea to update the docs there in a separate PR?

**Comment by mubariskhan96** [comment]
> The `fybrik` stuff also looks like it could be grouped, skimming through -- these seem like they ought to be grouped together:

I have updated the PR with the fybrik bundling changes

File: cluster-scope/overlays/prod/moc/smaug/kustomization.yaml (hunk 2 above)
Comment (HumairAK) [reply cluster-scope/overlays/prod/moc/smaug/kustomization.yaml:28]: kk sounds good!

**Pull request merged**